internal/clientconn/conninfo/conn_info.go

```diff
@@ -71,6 +73,14 @@ func (connInfo *ConnInfo) SetAuth(username, password string) {
 	connInfo.password = password
 }
 
+// SetClientMetadata sets the client metadata presence.
+func (connInfo *ConnInfo) SetClientMetadataPresence() {
+	connInfo.rw.Lock()
+	defer connInfo.rw.Unlock()
+
+	connInfo.ClientMetadataPresence = true
```

Comment:
If we set it under lock, then we should read it under lock. Otherwise, there is a race condition